src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -229,6 +230,11 @@ internal CookieContainer CookieContainer
             set
             {
                 CheckDisposedOrStarted();
+                if (value == null)
+                {
+                    throw new ArgumentNullException("CookieContainer", SR.net_http_invalid_cookiecontainer);
```

Comment:
Now, having said that, I will say that the behavior on Desktop is a little different.  On Desktop, we do throw a ArgumentNullException when trying to set a NULL value for CookieContainer. But in WinHttpHandler, we actually have a 3-state enum with CookieUsePolicy and the CookieContainer and we try to map that back to the HttpClientHandler logic.  I would suggest for now that you don't throw in the setter for CookieContainer but do throw during SendAsync() when you validate the combination of property values.